contracts/strategy/LiquidationStrategy.sol

```diff
@@ -230,180 +319,159 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
             }
-        } else {
-            // Condition #3 loss is allowed, so if there is loss, make this address pay for the loss.
-            uint256 minAmountOut = dexAmountOut < amountNeededToPayDebt ? dexAmountOut : amountNeededToPayDebt;
-            //if allowLoss is true, then the LiquidatorStrategy will pay the difference of amountNeededToPayDebt and dexAmountOut
-            uint256 fathomStablecoinReceived = _sellCollateral(
-                _vars.tokenAdapter.collateralToken(),
-                path,
-                _vars.router,
+            _depositStablecoin(amountNeededToPayDebt, _vars.liquidatorAddress);
+            idleCollateral[_collateralToken].collateralAmount += retrievedCollateralAmount;
+            idleCollateral[_collateralToken].amountNeededToPayDebt += amountNeededToPayDebt;
+            idleCollateral[_collateralToken].averagePriceOfCollateral = idleCollateral[_collateralToken].amountNeededToPayDebt.mul(WAD).div(idleCollateral[_collateralToken].collateralAmount);
+            emit LogFlashLiquidationSuccess(
+                _vars.liquidatorAddress,
+                amountNeededToPayDebt,
                 _collateralAmountToLiquidate,
-                minAmountOut
+                fathomStablecoinReceivedV2,
+                fathomStablecoinReceivedV3,
+                _vars.v2Ratio
             );
-
-            // If we didn't receive enough to repay the debt - send the amount we received
-            // and the liquidator will try to pay the difference with its own funds.
-            if (fathomStablecoinReceived < amountNeededToPayDebt) {
-                require(fathomStablecoin.balanceOf(address(this)) >= amountNeededToPayDebt, "flashLendingCall: not enough to repay debt");
+        } else {
+            uint256 _collateralAmountToLiquidateV2 = _vars.v2Ratio == 0 ? 0 : _collateralAmountToLiquidate.mul(_vars.v2Ratio).div(TENK);
+            uint256 balanceOfFXDBeforeSwap = fathomStablecoin.balanceOf(address(this));
+            if (_vars.v2Ratio > 0) {
+                fathomStablecoinReceivedV2 = _handleCollateralSellingV2(
+                    _vars,
+                    _collateralAmountToLiquidateV2,
+                    amountNeededToPayDebt.mul(_vars.v2Ratio).div(TENK)
+                );
+            }
+            if (_vars.v2Ratio < TENK) {
+                fathomStablecoinReceivedV3 = _sellCollateralV3(
+                    _collateralToken,
+                    address(fathomStablecoin),
+                    _collateralAmountToLiquidate.sub(_collateralAmountToLiquidateV2),
+                    uniswapV3Info[_vars.routerV3].poolFee,
+                    _vars.routerV3
+                );
+            }
+            if ((fathomStablecoinReceivedV2 + fathomStablecoinReceivedV3) < amountNeededToPayDebt) {
+                if (fathomStablecoin.balanceOf(address(this)) < amountNeededToPayDebt) {
+                    revert NotEnoughToRepayDebt();
+                }
             }
```

Comment:
can be replaced just with: 
 ```
if (fathomStablecoin.balanceOf(address(this)) < amountNeededToPayDebt) {
     revert NotEnoughToRepayDebt();
}
```
            part with `if ((fathomStablecoinReceivedV2 + fathomStablecoinReceivedV3) < amountNeededToPayDebt) {` is redundant here

Reply:
or even better to do this and avoid redundant contract call:

```
if (balanceOfFXDBeforeSwap +fathomStablecoinReceivedV2 + fathomStablecoinReceivedV3  < amountNeededToPayDebt) {
   revert NotEnoughToRepayDebt();
}
```